packages/web/nexus/subscription.ts

```diff
@@ -0,0 +1,322 @@
+import { MembershipSubscriptionPeriod, InputJsonValue, PrismaClient } from '@journaly/j-db-client'
+import {
+  arg,
+  booleanArg,
+  extendType,
+  objectType,
+  stringArg,
+} from '@nexus/schema'
+import stripe from './utils/stripe'
+
+const MembershipSubscription = objectType({
+  name: 'MembershipSubscription',
+  definition(t) {
+    t.model.id()
+    t.model.period()
+    t.model.userId()
+    t.model.expiresAt()
+    t.model.cancelAtPeriodEnd()
+    t.model.lastFourCardNumbers()
+    t.model.cardType()
+  }
+})
+
+const getSubscriptionPriceId = (subType: MembershipSubscriptionPeriod) => {
+  switch(subType) {
+    case MembershipSubscriptionPeriod.MONTHLY:
+      return 'price_1ISRgvB8OEjVdGPaQr7ZANW8'
+    case MembershipSubscriptionPeriod.QUARTERLY:
+      return 'price_1ISRgvB8OEjVdGPaeOx4m255'
+    case MembershipSubscriptionPeriod.ANNUALY:
+      return 'price_1ISRgvB8OEjVdGPam1PTr6hE'
+  }
+}
+
+const setPaymentMethod = async (
+  userId: number,
+  db: PrismaClient,
+  customerId: string,
+  paymentMethodId: string,
+  lastFourCardNumbers: string = '1234',
```

Comment:
We can retrieve the last four digits on the card by retrieving the payment method from stripe, instead of requiring the client to pass them in. I guess it's not the end of the world if they pass in bad info for some reason, but I think it would be less error prone to get the value straight from stripe.

Reply:
Just pushed up some changes for this. On the case where we were attaching a payment method I just grabbed these off of that response (within `setPaymentMethod`) and in the case where we're creating a new customer in Stripe I made a `retrieve` call after and throw if the card isn't present for some reason

Reply:
✅💰